File: exercism/python/word-count/word_count.py
import re

def word_count(phrase):
  words = re.sub("[,\s_]+", " ", phrase).lower().split(" ")
  freqs = dict()
  for word in words:
    clean_word = re.sub("(^\W+|\W+$)", "", word)
    if clean_word != "":
      freqs[clean_word] = freqs.setdefault(clean_word, 0) + 1
  return freqs

print word_count("hey,my_spacebar_is_broken.")

File: exercism/ruby/word-count/word_count.rb
class Phrase
  def initialize(phrase)
    @phrase = phrase
  end

  def word_count
    words = @phrase.downcase.split(/[ ,]+/).map do |x|
      x.gsub(/^\W+|\W+$/, "")
    end
    words.reduce(Hash.new(0)) do |totals, word|
      totals[word] += 1
      totals
    end
  end
end

module BookKeeping
  VERSION = 1
end